### PR TITLE
Vue component unit test isolation

### PR DESCRIPTION
**Pull request by benkelaar**
Fixes unit test isolation for Vue.js component unit tests

While working on a solution for #10786 I started to write a unit test for the `home.vue` component. That went fairly smoothly but after I had added it I noticed that stray events from the home.vue localVue I created ended up failing random other tests. Clearly test isolation was failing. I still haven't figured out exactly why. My standing theory is that the await on the `setValue(value)` somehow doesn't completely wait on the events that are being fired by dispatch because it's not marked as an asynchronous function itself. I think this lack of a-synchronicity somehow breaks the Promise chain. In any case adding the async keyword to the dispatch function seems to fix the test isolation issues.

I'm also tracking down the different stacktraces that are being thrown by Vue.js component unit tests, because they unnecessarily confuse investigations into what is happening. Shall I add those to this merge request or shall I create a new merge request for that?

### Changes
Most significant change is the addition of the `async` keyword to the `dispatch` function in `/src/libs/store/index.js`

Beyond that:
- Added unit test for `home.vue` component
- Small cleanup in `home.vue`
- Replaced use of `zone` by `utcOffset` in `/src/store/index.js`
- Couple other touchups.

----
UUID: d71d2c57-a73d-4591-b64d-e688584a9092

### CODE REVIEW

**Comment by benkelaar** [comment]
Did a quick check of the usages of dispatch. It looks like all usages either don't care about the return value of dispatch or already handle it via either `await` or promise stacking.

File: website/client/src/store/index.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const IS_TEST = process.env.NODE_ENV === 'test'; // eslint-disable-line no-proce
 let isUserLoggedIn = false;
 
 // eg, 240 - this will be converted on server as -(offset/60)
-const browserTimezoneOffset = moment().zone();
+const browserTimezoneOffset = -moment().utcOffset();
```
Comment (paglias) [inline website/client/src/store/index.js]: This probably works well but I think we'll switch all together in one go with the remaining usages of `.zone()` as described in https://github.com/HabitRPG/habitica/issues/10209

File: website/client/tests/unit/mixins/groupsUtilities.spec.js [hunk 2]
```diff
@@ -2,10 +2,13 @@ import Vue from 'vue';
 import groupsUtilities from '@/mixins/groupsUtilities';
 import { TAVERN_ID } from '@/../../common/script/constants';
 import generateStore from '@/store';
+import Store from '@/libs/store';
+
+Vue.use(Store);
```
Comment (paglias) [inline website/client/tests/unit/mixins/groupsUtilities.spec.js]: should this use a local Vue instance like in the other test?

```
const localVue = createLocalVue();
localVue.use(Store);
```

**Review by paglias — CHANGES_REQUESTED** [review]
left two comments but otherwise looks good 👍 It would also match Vuex's own behavior of always returning a promise from a call to dispatch

File: website/client/tests/unit/mixins/groupsUtilities.spec.js (hunk 2 above)
Comment (benkelaar) [reply website/client/tests/unit/mixins/groupsUtilities.spec.js]: Yeah I think so, but there's quite a varied amount of approaches in the different Vue.js component tests. Didn't want to open the pit of unifying them all. This however was required for me to get the tests to run at all. However, perhaps this is no longer necessary, will check.

File: website/client/src/store/index.js (hunk 1 above)
Comment (benkelaar) [reply website/client/src/store/index.js]: Cool, will revert for this one.

File: website/client/tests/unit/mixins/groupsUtilities.spec.js (hunk 2 above)
Comment (benkelaar) [reply website/client/tests/unit/mixins/groupsUtilities.spec.js]: Alright, got it to work with localVue. Indeed better.

**Comment by paglias** [comment]
thanks @benkelaar ! noted in your profile

**Pull request merged**